scripts/evaluate_performance_by_text_length.py

```diff
@@ -0,0 +1,92 @@
+"""
+Evaluate model for science tags on different text lengths
+to understand how performance changes by adding more text
+and whether the first words are more important than taking
+random words from the text
+"""
+from pathlib import Path
+import argparse
+import pickle
+import random
+import json
+
+from sklearn.pipeline import Pipeline
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.svm import SVC
+from sklearn.metrics import f1_score
+from sklearn.model_selection import train_test_split
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def evaluate_model_on_different_lengths(model, X_test, Y_test, random_sample=False):
+    results = []
+    for word_len in range(50, 300, 50):
+        X_test_subset = []
+        Y_test_subset = []
+        for i in range(len(X_test)):
+            x_test_words = X_test[i].split()
```

Comment:
I thought this evaluation would be on a random sentence/s rather than a random list of words. Not that a random list of words isn't also interesting, but it is a bit different since they are all jumbled up.
e.g.:
```
if random_sample:
    random.seed(42)
    start_i = random.randint(0,(len(x_test_words)-word_len))
    x_test_words = x_test_words[start_i:(start_i+word_len)]
```

Reply:
i.e. it takes away your ability to say the first sentence holds the most info, since your test is also a structured sentence vs an unstructured sentence

Reply:
the structure of the sentence is lost though after when doing tfidf. i am not 100% convinced but it is also something i have been thinking, do you reckon i should try it and see if we get different results? we can also add it as a param and produce yet another graph for the different way to randomise

Reply:
hey @lizgzil, do you want me to split on sentences as well in order to approve this?

Reply:
no that's ok, will approve 👍